### PR TITLE
Fix Async.* endpoints

### PR DESCRIPTION
**Pull request by philkra**
* `async_search.submit` request + response
* `async_search.get` request

### CODE REVIEW

File: specification/specs/x_pack/async_search/submit/AsyncSearchSubmitRequest.ts [hunk 1]
```diff
@@ -25,6 +25,10 @@
 interface AsyncSearchSubmitRequest extends RequestBase {
   path_parts?: {
     index?: Indices
+    batched_reduce_size?: long
+    wait_for_completion_timeout?: Time
+    keep_on_completion?: boolean
+    typed_keys?: boolean
```
Comment (delvedor) [inline specification/specs/x_pack/async_search/submit/AsyncSearchSubmitRequest.ts:31]: Are you sure about this? I think this should go in `query_parameters`.

Comment (philkra) [reply specification/specs/x_pack/async_search/submit/AsyncSearchSubmitRequest.ts:31]: Now that you mention it! Yes, they should be in `query_parameters`! The tests did pass though 😉

Comment (philkra) [reply specification/specs/x_pack/async_search/submit/AsyncSearchSubmitRequest.ts:33]: feedback addressed

Comment (delvedor) [reply specification/specs/x_pack/async_search/submit/AsyncSearchSubmitRequest.ts:33]: > The tests did pass though 

Yes, unfortunately, we can't check if a parameter should go in the URL or in the query :S

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**